### DISCUSSION TITLE
Compile Tasmota with Mac Air M1?

### DISCUSSION THREAD

**Discussion by Jason2866** [arendst/Tasmota#13175] [Q&A] [answered]
since the dev team has no Mac Air M1... I would like to know if it is possible to compile
Tasmota, Tasmota32, Tasmota32-S2 and Tasmota32-C3 with VSC/Platformio on a Mac M1?
Yes/No? Problems? Stuff to take care?

**Comment by TrDA-hab** [comment]
Necessarily compatible !!!
The problem can only be with:
-  your USB-TTL module (or its software driver).
-  or your software (VSC/Platformio/...).
-  or in the COM port number (I already have such problems on Tasmota + Win10 + Platformio).

**Reply by martin072** [reply]
Have an M1 and no issues so far, Arduino, Visual Code, Platform IO work ok. Indeed with Big Sur some issues with the FTDI driver, but that was fixed (for this who need it: https://www.ftdichip.com/Drivers/VCP/MacOSX/FTDIUSBSerialDextInstaller_1_4_7.zip)

**Comment by Jason2866** [comment]
@martin072 Compiled the Tasmota builds for C3 and S2 too? All needed toolchains there?

**Reply by martin072** [reply]
I only use D1 Mini's and some sonoff basics. use the standard builds with some options enabled

**Reply by Jason2866** [reply]
Could you try to build the variants S2 and C3? You only have to rename `platformio_tasmota_cenv_sample.ini` to `platformio_tasmota_cenv.ini` 
and `platformio_override.ini_sample.ini ` `platformio_override.ini `
and uncomment https://github.com/arendst/Tasmota/blob/development/platformio_override_sample.ini#L36-L37. Restart VSC.
Big THX

**Comment by martin072** [comment]
Compression reduced firmware size by 71% (was 714960 bytes, now 510018 bytes)
====================================== [SUCCESS] Took 18.60 seconds ======================================

Environment    Status    Duration
-------------  --------  ------------
tasmota        SUCCESS   00:00:18.598
======================================= 1 succeeded in 00:00:18.598 =======================================

**Reply by martin072** [reply]
No issues

**Reply by martin072** [reply]
Sorry that was for the esp8266, not the esp32. Will try again

**Reply by martin072** [reply]
Hmz, fails with this error: Tasmota-development-9.5.0.8/tasmota/xdrv_17_rcswitch.ino:43:10: fatal error: RCSwitch.h: No such file or directory

Not sure if this has to do with the M1, I did do some ESP32 builds, and for sure the esp8266 builds work fine...

**Comment by Jason2866** [comment, marked as answer]
Thx, for trying. The S2 and C3 do build without errors on Windows, Linux and Mac OS (Intel x86).
The error you encounter with `RCSwitch.h` looks like a general error in your setup. Probably you have personal settings done in `platformio_override.ini` which causing issues. Or it is indeed a strange error caused by the M1 :- (

**Reply by Jason2866** [reply]
Compile does work using Rosetta and it is fast!
